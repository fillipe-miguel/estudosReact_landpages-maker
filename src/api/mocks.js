/* eslint-disable import/prefer-default-export */
export const pageDataMock = [
  {
    title: 'Title teste 01',
    slug: 'title-teste-01',
    menu: {
      logo_link: '#link-teste',
      logo_text: 'LOGO TESTE',
      logo: {
        data: {
          attributes: {
            url: 'https://url-da-imagem_teste.svg',
          },
        },
      },
      menu_link: [
        {
          link_text: 'Text do link test',
          url: '#link-test',
          open_in_new_tab: true,
        },
      ],
    },
    sections: [
      // ================================= Layout grid column / section two columns =================================
      {
        __component: 'section.section-two-columns',
        title: 'Essa é a seção de duas colunas',
        description: 'Descrição da seção de duas colunas',
        image: {
          data: {
            attributes: {
              url: 'https://layoutgridcolumn-test.jpg',
            },
          },
        },
        metadata: {
          section_id: 'section-two-columns',
          background: false,
        },
      },

      // ================================= Layout center content / section content ==================================
      {
        __component: 'section.section-content',
        title: 'Essa é a seção de uma coluna e só texto',
        content: 'lorem lorem lorem lorem lorem',
        metadata: {
          section_id: 'section-center-content',
          background: true,
        },
      },

      // ================================= Layout grid describe/ section grid text =================================
      {
        __component: 'section.section-grid',
        title: 'Aqui é a seção de grid com texto',
        description: 'Vários textos ai olha que loucura!',
        metadata: {
          section_id: 'grid-text',
          background: true,
        },
        text_grid: [
          {
            title: 'Teste 1',
            description: 'Lorem Ipsum teste',
          },
          {
            title: 'Teste 2 ',
            description:
              "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
          },
          {
            title: 'Teste 3',
            description:
              "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
          },
        ],
        image_grid: [],
      },

      // ================================= Section grid image / Layout grid gallery =================================
      {
        __component: 'section.section-grid',
        title: 'Section grid com imagens / gallery',
        description: 'Éssa é a seção com o grid de imagens muito loucas',
        metadata: {
          name: 'section-grid-gallery',
          section_id: 'section-grid-gallery',
          background: true,
        },
        text_grid: [],
        image_grid: [
          {
            image: {
              data: {
                attributes: {
                  alternativeText: 'Mulher segurando o peso',
                  url: 'https://section-grid-gallery.jpg',
                },
              },
            },
          },
        ],
      },
    ],
    footer: '<p>Teste aqui</p>',
  },
];
