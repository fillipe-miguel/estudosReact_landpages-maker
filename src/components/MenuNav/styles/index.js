import styled, { css } from 'styled-components';

export const MenuNavRoot = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;

    @media ${theme.medias.lteMedium} {
      flex-flow: column nowrap;
      align-items: center;
    }
  `}
`;

export default MenuNavRoot;
