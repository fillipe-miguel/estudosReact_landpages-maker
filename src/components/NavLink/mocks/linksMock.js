export default [
  {
    link: 'https://www.google.com.br',
    children: 'Google',
    newTab: true,
  },
  {
    link: 'https://youtube.com',
    children: 'YouTube',
    newTab: true,
  },
  {
    link: 'https://github.com/fillipe-miguel',
    children: 'My GitHub',
    newTab: false,
  },
  {
    link: 'https://www.tiktok.com/pt-BR/',
    children: 'TikTok',
    newTab: true,
  },
  {
    link: 'https://www.udemy.com/',
    children: 'Udemy',
    newTab: true,
  },
];
