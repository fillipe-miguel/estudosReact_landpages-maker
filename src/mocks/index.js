import layouts from './layouts';
import components from './components';

export { layouts, components };
