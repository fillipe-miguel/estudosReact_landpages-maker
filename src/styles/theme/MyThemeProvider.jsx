import P from 'prop-types';
import React from 'react';
// Styles
import { ThemeProvider } from 'styled-components';
import GlobalStyles from '../global-styles';
import theme from './theme';

function MyThemeProvider({ children }) {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
}

MyThemeProvider.propTypes = {
  children: P.node.isRequired,
};

export default MyThemeProvider;
