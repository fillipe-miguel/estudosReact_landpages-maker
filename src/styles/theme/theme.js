const theme = {
  // Your theme HERE
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#DC143C',

    lightBg: '#FFF',
    darkBg: '#0A1128',

    white: '#FFF',
    black: '#000',
  },
  fonts: {
    sizes: {
      xsmall: '8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4rem',
      xxlarge: '4.8rem',
      // Pulei um pois achei que n√£o usaria !
      xxxlarge: '6.4rem',
    },
  },
  spacings: {
    xsmall: '0.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4rem',
    xxlarge: '4.8rem',
  },
  medias: {
    lteMedium: '(max-width: 768px)',
  },
};

export default theme;
