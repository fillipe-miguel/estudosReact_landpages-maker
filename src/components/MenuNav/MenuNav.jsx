import P from 'prop-types';
import React from 'react';
// Components
import MenuLink from '../MenuLink';
// Styles
import * as Styled from './styles';

function MenuNav({ links }) {
  return (
    <Styled.MenuNavRoot>
      {links.map((link) => (
        <MenuLink key={link.link} link={link.link} newTab={link?.newTab}>
          {link.children}
        </MenuLink>
      ))}
    </Styled.MenuNavRoot>
  );
}

MenuNav.defaultProps = {
  links: [],
};

MenuNav.propTypes = {
  links: P.arrayOf(
    P.shape({
      link: P.string.isRequired,
      children: P.string.isRequired,
      newTab: P.bool,
    }),
  ),
};

export default MenuNav;
