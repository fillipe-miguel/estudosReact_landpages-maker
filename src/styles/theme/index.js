import MyThemeProvider from './MyThemeProvider';
import theme from './theme';

export { MyThemeProvider, theme };
