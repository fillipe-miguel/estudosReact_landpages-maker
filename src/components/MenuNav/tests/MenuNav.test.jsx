/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';
// Styles
import MyThemeProvider from '../../../styles/theme/MyThemeProvider';
import theme from '../../../styles/theme/theme';
// Component
import MenuNav from '..';
// Mocks
import { linksMock } from '../../../mocks/componentsMocks';

describe('<MenuNav />', () => {
  // ================================= Teste 1 =================================
  it('should render', () => {
    render(
      <MyThemeProvider>
        <MenuNav links={linksMock} />
      </MyThemeProvider>,
    );

    expect(screen.getByRole('navigation')).toBeInTheDocument();
  });

  // ================================= Teste 2 =================================
  it('should render without links', () => {
    render(
      <MyThemeProvider>
        <MenuNav />
      </MyThemeProvider>,
    );

    expect(screen.queryAllByRole('link')).toHaveLength(0);
  });

  // ================================= Teste 3 =================================
  it('should render with all links', () => {
    render(
      <MyThemeProvider>
        <MenuNav links={linksMock} />
      </MyThemeProvider>,
    );

    const container = screen.getAllByRole('link');

    expect(container).toHaveLength(linksMock.length);
  });

  // ================================= Teste 4 =================================
  it('should render with screen mobile', () => {
    render(
      <MyThemeProvider>
        <MenuNav links={linksMock} />
      </MyThemeProvider>,
    );

    const container = screen.getByRole('navigation');

    expect(container).toHaveStyleRule('flex-flow', 'column nowrap', {
      media: theme.medias.lteMedium,
    });
  });
  // ================================= SNAPSHOT  =================================
  it('should match snapshot', () => {
    const { container } = render(
      <MyThemeProvider>
        <MenuNav links={linksMock} />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
