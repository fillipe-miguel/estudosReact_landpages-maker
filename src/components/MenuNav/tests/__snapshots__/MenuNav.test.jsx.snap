// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuNav /> should match snapshot 1`] = `
.c1 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 1rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #DC143C;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<nav
  aria-label="Main menu"
  class="c0"
>
  <a
    class="c1"
    href="https://www.google.com.br"
    target="_blank"
  >
    Google
  </a>
  <a
    class="c1"
    href="https://youtube.com"
    target="_blank"
  >
    YouTube
  </a>
  <a
    class="c1"
    href="https://github.com/fillipe-miguel"
    target="_self"
  >
    My GitHub
  </a>
  <a
    class="c1"
    href="https://www.tiktok.com/pt-BR/"
    target="_blank"
  >
    TikTok
  </a>
  <a
    class="c1"
    href="https://www.udemy.com/"
    target="_blank"
  >
    Udemy
  </a>
</nav>
`;
